modules/oe_translation_cdt/src/Mapper/TranslationRequestMapper.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_translation_cdt\Mapper;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Site\Settings;
+use Drupal\Core\Url;
+use Drupal\oe_translation\TranslationSourceHelper;
+use Drupal\oe_translation_cdt\ContentFormatter\ContentFormatterInterface;
+use Drupal\oe_translation_cdt\TranslationRequestCdtInterface;
+use OpenEuropa\CdtClient\Model\Request\Callback;
+use OpenEuropa\CdtClient\Model\Request\File;
+use OpenEuropa\CdtClient\Model\Request\SourceDocument;
+use OpenEuropa\CdtClient\Model\Request\Translation;
+use OpenEuropa\CdtClient\Model\Request\TranslationJob;
+
+/**
+ * Maps the data between the TranslationRequestCDT entity and the DTO.
+ */
+class TranslationRequestMapper implements DtoMapperInterface {
+
+  protected const CHARACTERS_PER_PAGE = 750;
+
+  protected const VOLUME_MULTIPLIER = 0.5;
+
+  /**
+   * TranslationRequestMapper constructor.
+   *
+   * @param \Drupal\oe_translation_cdt\ContentFormatter\ContentFormatterInterface $contentFormatter
+   *   The content formatter.
+   */
+  public function __construct(
+    protected ContentFormatterInterface $contentFormatter
+  ) {}
+
+  /**
+   * Converts Drupal TranslationRequest entity to CDT library DTO.
+   *
+   * @param \Drupal\oe_translation_cdt\TranslationRequestCdtInterface $entity
+   *   The translation request entity.
+   *
+   * @return \OpenEuropa\CdtClient\Model\Request\Translation
+   *   The translation request DTO.
+   */
+  public function convertEntityToDto(ContentEntityInterface $entity): Translation {
```

Comment:
There is a mismatch between what is in annotation and what is set as parameter type here.
Same for convertEntityToDto vs createSourceDocument, different param types even though they get the same entity object.

Reply:
This is because I planned to introduce a more general interface for DTO mappers. But maybe it doesn't make sense in this case, as we only have one mapper. I will change it.